pkg/asset/agent/installconfig.go

```diff
@@ -29,9 +47,129 @@ func (a *OptionalInstallConfig) Generate(parents asset.Parents) error {
 
 // Load returns the installconfig from disk.
 func (a *OptionalInstallConfig) Load(f asset.FileFetcher) (bool, error) {
-	found, err := a.InstallConfig.Load(f)
+
+	var found bool
+
+	// First load the provided install config to early validate
+	// as per agent installer specific requirements
+	// Detailed generic validations of install config are
+	// done by pkg/asset/installconfig/installconfig.go
+	installConfig, err := a.loadEarly(f)
+	if err != nil {
+		return found, err
+	}
+
+	if err := a.validateInstallConfig(installConfig).ToAggregate(); err != nil {
+		return found, errors.Wrapf(err, "invalid install-config configuration")
+	}
+
+	found, err = a.InstallConfig.Load(f)
 	if found && err == nil {
 		a.Supplied = true
 	}
 	return found, err
 }
+
+// loadEarly loads the install config from the disk
+// to be able to validate early for agent installer
+func (a *OptionalInstallConfig) loadEarly(f asset.FileFetcher) (*types.InstallConfig, error) {
+
+	file, err := f.FetchByName(installConfigFilename)
+	config := &types.InstallConfig{}
+	if err != nil {
+		if os.IsNotExist(err) {
+			return config, nil
+		}
+		return config, errors.Wrap(err, asset.InstallConfigError)
+	}
+
+	if err := yaml.UnmarshalStrict(file.Data, config, yaml.DisallowUnknownFields); err != nil {
+		if strings.Contains(err.Error(), "unknown field") {
+			err = errors.Wrapf(err, "failed to parse first occurence of unknown field")
+		}
+		err = errors.Wrapf(err, "failed to unmarshal %s", installConfigFilename)
+		return config, errors.Wrap(err, asset.InstallConfigError)
+	}
+	return config, nil
+}
+
+func (a *OptionalInstallConfig) validateInstallConfig(installConfig *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
```

Comment:
Nit - would be good to use same definition of allErrs as the other functions, i.e.g:
`var allErrs field.ErrorList`